editor/src/clj/editor/gui.clj

```diff
@@ -810,8 +1091,123 @@
   (output node-overrides g/Any :cached (g/fnk [node-overrides id _overridden-properties]
                                          (into {id _overridden-properties}
                                                node-overrides)))
+  (input layout-names g/Any)
+  (output layout-names g/Any (gu/passthrough layout-names))
```

Comment:
Layout names are passed from scene to all nodes so we can figure out where a layout was introduced along the override chain.